File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        curr_sum = 0
        for i in range(k):
            curr_sum += nums[i]
        max_sum = curr_sum
        L = 0
        for R in range(k, len(nums)):
            curr_sum += nums[R]
            if R - L + 1 > k:
                curr_sum -= nums[L]
                L += 1
            max_sum = max(max_sum, curr_sum)
        return max_sum / k

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double curr = 0, res;
        for (int i = 0; i < k; ++i) {
            curr += nums[i];
        }

        res = curr;
        for (int i = k; i < nums.size(); ++i) {
            curr += (nums[i] - nums[i - k]);
            res = max(res, curr);
        }

        return res / k;
    }
};